lib/runtime/offchain/httpset.go

```diff
@@ -45,10 +49,35 @@ func (b requestIDBuffer) put(i int16) error {
 	}
 }
 
+// Request holds the request object and update the invalid and waiting status whenever
+// the request starts or is waiting to be read
+type Request struct {
+	Request          *http.Request
+	invalid, waiting bool
```

Comment:
is this concurrent safe?

Reply:
also, when are these variables set and what are they used for?

Reply:
for now, there is no concurrency (will be implemented here #2020). The `invalid` and `waiting` are set to `true` when we're waiting for a response, then the `waiting` is set to `false` when we got a successful or failed response however `invalid` remains `true`

Reply:
I don't see where in the code they're set (unless it wasn't in this PR?)

Reply:
Yes, right now these values aren't being set because the function that does this is not implemented yet!